### PR TITLE
Protect against and diagnose OrbitReset issues

### PR DESCRIPTION
**Pull request by mpuccio**
This change adds the caching of the start-of-run and end-of-run timestamps
in addition to the orbit-reset timestamp. It also validates that the
calculated timestamp is within the run duration before adding it to the
timestamp table. This ensures that only valid timestamps are used,
improving the reliability of the data.

### CODE REVIEW

File: Common/TableProducer/timestamp.cxx [hunk 1]
```diff
@@ -75,11 +77,12 @@ struct TimestampTask {
...
       LOGF(debug, "Getting orbit-reset timestamp from cache");
       orbitResetTimestamp = mapRunToOrbitReset[runNumber];
+      runDuration = mapRunToRunDuration[runNumber];
     } else { // The run was not requested before: need to acccess CCDB!
       LOGF(debug, "Getting start-of-run and end-of-run timestamps from CCDB");
-      auto timestamps = ccdb->getRunDuration(runNumber, true); /// fatalise if timestamps are not found
-      int64_t sorTimestamp = timestamps.first;                 // timestamp of the SOR in ms
-      int64_t eorTimestamp = timestamps.second;                // timestamp of the EOR in ms
+      runDuration = ccdb->getRunDuration(runNumber, true); /// fatalise if timestamps are not found
+      int64_t sorTimestamp = runDuration.first;            // timestamp of the SOR in ms
```
Comment (chiarazampolli) [inline Common/TableProducer/timestamp.cxx:84]: This is not necessarily from SOR, and below not from EOR. I would update the comment.

File: Common/TableProducer/timestamp.cxx [hunk 2]
```diff
@@ -104,14 +107,18 @@ struct TimestampTask {
...
     }
 
     if (verbose.value) {
       LOGF(info, "Orbit-reset timestamp for run number %i found: %llu us", runNumber, orbitResetTimestamp);
     }
-
-    timestampTable((orbitResetTimestamp + int64_t(bc.globalBC() * o2::constants::lhc::LHCBunchSpacingNS * 1e-3)) / 1000); // us -> ms
+    int64_t timestamp{(orbitResetTimestamp + int64_t(bc.globalBC() * o2::constants::lhc::LHCBunchSpacingNS * 1e-3)) / 1000}; // us -> ms
+    if (timestamp < runDuration.first || timestamp > runDuration.second) {
+      LOGF(warn, "Timestamp %llu us is out of run duration [%llu, %llu] ms", timestamp, runDuration.first, runDuration.second);
```
[the remark this answers is not in the document]
Comment (shahor02) [reply Common/TableProducer/timestamp.cxx:119]: Why fatalize? I would simply assign the runDuration.first, then you don't lose the whole analysis due to the bogus empty TF created at the end of run. As far as I can see, no object refers to such a timestamp, apparently they were added as "first orbit" of bogus TFs.

**Comment by mpuccio** [comment]
Hi @chiarazampolli : I would still keep the same behaviour to avoid issues with the analysers, so I put a configurable that regulates the "fatalisation" on invalid BCs. As Ruben saw in the AO2Ds for the other run, we might have a few spurious BCs, but if they are not referred by other tables there is no point in issuing a `fatal` here. The users of the timestamp should be aware and avoid relying blindly on this information, as it might be corrupted (see the fix to the event selection in #7618 )

**Review by jgrosseo — COMMENTED** [review]
For me this is fine.

@ddobrigk Any comments from your side?

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**